components/uniset/src/enum_props.rs

```diff
@@ -16,249 +18,304 @@ pub enum EnumeratedProperty {
     ScriptExtensions = 0x7000,
 }
 
+/// Enumerated Unicode general category types.
+/// GeneralSubcategory only supports specific subcategories (eg UppercaseLetter).
+/// It does not support grouped categories (eg Letter). For grouped categories, use GeneralCategory.
+#[derive(Copy, Clone, PartialEq, Debug, TryFromPrimitive, UnsafeFromPrimitive)]
+#[allow(missing_docs)] // TODO(#1030) - Add missing docs.
+#[repr(u8)]
+pub enum GeneralSubcategory {
+    Unassigned = 0,
+
+    UppercaseLetter = 1,
+    LowercaseLetter = 2,
+    TitlecaseLetter = 3,
+    ModifierLetter = 4,
+    OtherLetter = 5,
+
+    NonspacingMark = 6,
+    EnclosingMark = 7,
+    SpacingMark = 8,
+
+    Digit = 9,
+    LetterNumber = 10,
+    OtherNumber = 11,
+
+    SpaceSeparator = 12,
+    LineSeparator = 13,
+    ParagraphSeparator = 14,
+
+    Control = 15,
+    Format = 16,
+    PrivateUse = 17,
+    Surrogate = 18,
+
+    DashPunctuation = 19,
+    OpenPunctuation = 20,
+    ClosePunctuation = 21,
+    ConnectorPunctuation = 22,
+    OtherPunctuation = 23,
+    InitialPunctuation = 28,
+    FinalPunctuation = 29,
+
+    MathSymbol = 24,
+    CurrencySymbol = 25,
+    ModifierSymbol = 26,
+    OtherSymbol = 27,
+}
+
 /// Enumerated Unicode general category types.
 /// The discriminants correspond to the U_GC_XX_MASK constants in ICU4C.
-/// This supports groups of general categories: for example, `Letter`
+/// Unlike GeneralSubcategory, this supports groups of general categories: for example, `Letter`
 /// is the union of `UppercaseLetter`, `LowercaseLetter`, etc...
 /// See https://www.unicode.org/reports/tr44/ .
 /// See UCharCategory and U_GET_GC_MASK in ICU4C.
 #[derive(Clone, PartialEq, Debug)]
 #[allow(missing_docs)] // TODO(#1030) - Add missing docs.
 #[repr(u32)]
-#[non_exhaustive]
```

Comment:
Nit 1: I think you should keep the `#[non_exhaustive]`

Nit 2 (optional): I would argue that this isn't really an enum; it's a set of **selectors** for common cases.  If it were me, these would be constants wrapped in a newtype.  But I don't feel strongly enough on that point to hold up the PR.

Reply:
It's true that in principle we could have a different implementation that allows users to select custom sets of subcategories (say, Letter+Number). However: 
1. Our current data model - that is, the way we define our resource keys - can't provide data for a selector we haven't defined. 
2. I think that usecase is easily satisfied by just making multiple queries.
3. If we keep this marked as non-exhaustive, we can switch later.

Reply:
I went to write `impl From<GeneralSubcategory> for GeneralCategory`, and now I lean towards making this a newtype-wrapped selector, because it means the `from` impl is just a bitshift instead of a match.